pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -328,11 +328,11 @@ func getOptimalBlockIOForDevice(path string) (*api.BlockIO, error) {
 	}
 	defer util.CloseIOAndCheckErr(f, nil)
 
-	logicalSize, err := unix.IoctlGetInt(int(f.Fd()), unix.BLKSSZGET)
+	logicalSize, err := unix.IoctlGetUint32(int(f.Fd()), unix.BLKSSZGET)
```

Comment:
Doesn't kernel expect int? Forcing it to 32 might still work but we should, preferably use what is expected.

Reply:
@xpivarc when invoking unix.IoctlGetInt() for BLKSSZGET which internally invoke ioctlPtr() which taking 3rd argument of type int (In Go it will be 8 bytes), But underneath when it invokes system call under supervisor mode which follows C standard ABI, which consider it as a 4 Bytes. When it is updating data at the specified address it updating upper 4-Bytes i.e, starting 4-bytes because it is considering it as 4 bytes. This causes problem only in big-endian platform, But for little endian it will not harm, even-though "int" size mismatching.

Reply:
@xpivarc when we invoke unix.IoctlGetInt() for BLKSSZGET which internally invoke ioctlPtr() which taking 3rd argument of type int (In Go it will be 8 bytes), But underneath when it invokes system call under supervisor mode which follows C standard ABI, which consider it as a 4 Bytes. When it is updating data at the specified address it updating upper 4-Bytes i.e, starting 4-bytes because it is considering it as 4 bytes. This causes problem only in big-endian platform, But for little endian it will not harm, even-though "int" size mismatching.

Reply:
You are right, I am too used to int being platform specific.